sphinx/transforms/post_transforms/code.py

```diff
@@ -96,21 +103,61 @@ def visit_literal_block(self, node):
             node['linenos'] = (lines >= setting.lineno_threshold - 1)
 
 
+class HighlightLanguageDetector(SphinxTransform):
```

Comment:
To determine language on post transform phase is good idea. But this implementation conceals what language is specified by user. On highlighting, highlighter does not emit warnings for `default` language even if error. We should consider such case.

Reply:
I'll try this in another PR.